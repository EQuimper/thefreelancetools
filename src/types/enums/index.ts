export * from './ModalTypeEnum';
export * from './SidebarIconEnum';
