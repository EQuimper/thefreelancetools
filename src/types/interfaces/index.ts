export * from './ThemeInterface';
export * from './ElapsedTimeInterface';
