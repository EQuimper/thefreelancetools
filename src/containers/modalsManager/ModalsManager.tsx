import { Dialog } from 'evergreen-ui';
import { observer } from 'mobx-react';
import * as React from 'react';

import { NewProjectModal } from '@freelance-tool/components';
import { store } from '@freelance-tool/models';

import { ModalTypeEnum } from '../../types';
// import styled from 'styled-components';

// const Root = styled.div`
//   display: grid;
// `;

interface P {}

interface S {}

@observer
class ModalsManager extends React.Component<P, S> {
  state = {};

  _renderModal = () => {
    switch (store.modalsManager.modalType) {
      case ModalTypeEnum.NEW_PROJECT:
        return <NewProjectModal />;
      default:
        // Need to pass a div cause Dialog required a children
        return <div />;
    }
  }
  render() {
    const { modalsManager } = store;
    return (
      <Dialog
        isShown={modalsManager.isShow}
        title={modalsManager.modalTitle}
        hasHeader={!!modalsManager.modalTitle}
        onCloseComplete={modalsManager.close}
        confirmLabel={modalsManager.confirmButtonLabel}
        hasFooter={!!modalsManager.confirmButtonLabel}
      >
        {this._renderModal()}
      </Dialog>
    );
  }
}

export default ModalsManager;
