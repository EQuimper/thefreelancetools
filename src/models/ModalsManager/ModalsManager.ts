import { types } from 'mobx-state-tree';

import { ModalTypeEnum } from '../../types';

export const ModalsManager = types
  .model('ModalsManager', {
    isShow: false,
    modalType: types.maybe(types.enumeration('modalType', ['newProject'])),
  })
  .actions(self => ({
    open(modalType: ModalTypeEnum) {
      self.isShow = true;
      self.modalType = modalType;
    },
  }))
  .actions(self => ({
    close() {
      self.isShow = false;
      self.modalType = null;
    },
  }));
