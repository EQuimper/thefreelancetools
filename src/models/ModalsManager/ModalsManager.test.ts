import { onSnapshot } from 'mobx-state-tree';

import { ModalTypeEnum } from '../../types';
import { ModalsManager } from './ModalsManager';

describe('ModalsManager model', () => {
  it('should create a instance of a model', () => {
    const modalsManager = ModalsManager.create();

    expect(modalsManager.isShow).toBe(false);
    expect(modalsManager.modalType).toBe(null);
  });

  it('should be able to open a modal with the modalType newProject', () => {
    const modalsManager = ModalsManager.create();

    const states: any[] = [];

    onSnapshot(modalsManager, snapshot => {
      states.push(snapshot);
    });

    modalsManager.open(ModalTypeEnum.NEW_PROJECT);

    expect(modalsManager.isShow).toBe(true);
    expect(modalsManager.modalType).toBe('newProject');

    expect(states).toMatchSnapshot();
  });

  it('should be able to close a modal', () => {
    const modalsManager = ModalsManager.create();

    const states: any[] = [];

    onSnapshot(modalsManager, snapshot => {
      states.push(snapshot);
    });

    modalsManager.open(ModalTypeEnum.NEW_PROJECT);

    modalsManager.close();

    expect(modalsManager.isShow).toBe(false);
    expect(modalsManager.modalType).toBe(null);

    expect(states).toMatchSnapshot();
  });
});
