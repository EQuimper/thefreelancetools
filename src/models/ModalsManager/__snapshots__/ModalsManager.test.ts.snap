// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalsManager model should be able to close a modal 1`] = `
Array [
  Object {
    "isShow": true,
    "modalType": "newProject",
  },
  Object {
    "isShow": false,
    "modalType": null,
  },
]
`;

exports[`ModalsManager model should be able to open a modal with the modalType newProject 1`] = `
Array [
  Object {
    "isShow": true,
    "modalType": "newProject",
  },
]
`;
